pkg/cli/tail.go

```diff
@@ -176,21 +200,50 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("-max-num must be greater than 0 and less than 1000")
 	}
 
-	// Tail logs.
-	ls, err := c.tail(ctx)
+	var extra []validation.Validator
+	if c.flagAdditionalCheck {
+		extra = append(extra, validation.ValidateLabels)
+	}
+
+	var puller logPuller
+	if c.testPuller != nil {
+		puller = c.testPuller
+	} else {
+		logClient, err := logging.NewClient(ctx)
+		defer func() {
+			if err := logClient.Close(); err != nil {
+				rErr = errors.Join(rErr, fmt.Errorf("failed to close log client: %w", err))
+			}
+		}()
+		if err != nil {
+			return errors.Join(rErr, fmt.Errorf("failed to create logging client: %w", err))
+		}
+		puller = cloudlogging.NewPuller(ctx, logClient, c.flagScope, cloudlogging.WithRetry(retry.WithMaxRetries(uint64(c.flagFollowDuration), retry.NewConstant(1000*time.Millisecond))))
+	}
+
+	if c.flagFollow {
```

Comment:
Should we do some checks to prevent flag conflicts? like when both -f and -n are provided. Also duration flag is useless in stream tail, we can add usage explanation or some other way to make that clear.

Reply:
Just update the usage explanation for duration flag that it does not change the outcome if it is stream tail.

Reply:
Updated the usage for flag `-n` and `-d`